pandas/core/arrays/base.py

```diff
@@ -245,3 +255,11 @@ def _can_hold_na(self):
         Setting this to false will optimize some operations like fillna.
         """
         return True
+
+    def value_counts(self, dropna=True):
+        from pandas import value_counts
+
+        if dropna:
+            self = self[~self.isna()]
+
+        return value_counts(np.array(self))
```

Comment:
This is too tied to materializing using ndarary. rather this should raise AbstractMethodError

Reply:
We may have a difference in philosophy here then. I'm trying to follow the lead of standard library in providing default implementations for as much as possible, even if they're sub-optimal.

Reply:
What's strange about that? The fact that it's being added, or that it's returning a Series? I'm not sure what else it would return.

Ideally, we'll someday have an `Index[extension_dtype]`, so that you get a Series with an exetension-array backed index. That's seems like the most natural return value.

Reply:
you need to wrap this on the return, IOW call EA(....) here

Reply:
> you need to wrap this on the return, IOW call EA(....) here

That won't work as the result is a Series

> What's strange about that? The fact that it's being added, or that it's returning a Series? I'm not sure what else it would return.

The fact that it is a Series (a Series return value from array method 'feels' wrong, as the hierarchy is the other way around, but probably it is just me :-))

Reply:
I *think* we're OK since `pd.value_counts(array)` *always* returns a series. `Categorical.value_counts` returns a Series. The only special thing is the index type which would be an `ExtensionIndex` in an ideal world, but is an `Index[object]` for now.

Reply:
hmm, not real happy with this interface, but I guess ok for now. let's revisit this at somepoint in the future.

Reply:
@TomAugspurger I suppose you want this in here, so you can implement a more efficient method on IPArray? 
Because we could also, for now, keep this implementation in `pd.value_counts` (so for people who want this functionality, they can use it) instead of adding it as a method on ExtensionArray interface itself. And we can revisit when talking about unique/factorize/duplicated/.. algos support for ExtensionArrays in general?

Reply:
I mostly added it since that's what `Categorical` does. I can see if removing it breaks anything.

Reply:
Just to make sure, we're OK with `pd.value_counts` and `Series[EA]` currently failing for 3rd party EAs? By removing that from the interface, we'll see

```pytb
AttributeError: 'DecimalArray' object has no attribute 'value_counts'
```

I could make that error message better, but I'd prefer waiting on that till we flesh out what we want `pd.value_counts(EA)` to do.

Reply:
But in principle you can easily add this fallback to do a value_counts on the array of objects there?